### PR TITLE
Added InsecureSkipVerify flag

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
## Read then delete

# TL;DR
This is context of the Flyte EKS deployment where we want to allow users to deploy a self signed cert for playing with the flyte environment.

On secure connection in a test environment its usually easier to deploy self signed certs and hence don't require CA trust , hostname validation.

This is similar to behaviour in browser which show you a warning if you are visiting a secure page but has invalid certificate due to various reasons mentioned above.

Adding this flag is only for test environments and the user is cautioned about it usage in the documentation.

Without the flag
```
flytectl get project   
Error: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: x509: certificate is valid for flyte.example.org, not k8s-flyte-8699360f2e-2074544761.us-east-2.elb.amazonaws.com"
```

With the flag defined in config.yaml

```
flytectl get project                            
 ---------------- ---------------- --------------------------- 
| ID (4)         | NAME           | DESCRIPTION               |
 ---------------- ---------------- --------------------------- 
| flyteexamples  | flyteexamples  | flyteexamples description |
 ---------------- ---------------- --------------------------- 
| flytesnacks    | flytesnacks    | flytesnacks description   |
 ---------------- ---------------- --------------------------- 
| flytetester    | flytetester    | flytetester description   |
 ---------------- ---------------- --------------------------- 
| myflyteproject | myflyteproject |                           |
 ---------------- ---------------- --------------------------- 
```

flyte-cli and any other interfaces should also be updated to use the new flag which access flyte from ingress endpoint

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [X] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: clients/go/admin/client.go [hunk 1]
```diff
@@ -196,7 +197,16 @@ func NewAdminConnection(_ context.Context, cfg *Config, opts ...grpc.DialOption)
 		opts = append(opts, grpc.WithInsecure())
 	} else {
 		// TODO: as of Go 1.11.4, this is not supported on Windows. https://github.com/golang/go/issues/16736
-		creds := credentials.NewClientTLSFromCert(nil, "")
+		var creds credentials.TransportCredentials
+		if cfg.InsecureSkipVerify {
```
[the remark this answers is not in the document]
Comment (pmahindrakar-oss) [reply clients/go/admin/client.go:201]: Done

**Review by EngHabu — APPROVED** [review]

**Review by kumare3 — APPROVED** [review]
